### ISSUE TITLE
[stack upgrade] Errors when upgrading rom 6.5 to 7.0 with Metricbeat data

### ISSUE THREAD

**Issue by cjcenizal** [elastic/beats#11385]
My test case is that I'm using Metricbeat to ingest data through Logstash and also using Metricbeat's sample dashboards, starting on 6.5. I upgrade to 6.7 and then 7.0.

## Upgrading from 6.5 to 6.7

Overall, this process went smoothly. The only special consideration I had to make was that since I was using Logstash to rename the documents sent from Metricbeat, I had to deliberately change the Logstash naming pattern to ensure that ingested documents were captured by Metricbeat's new index template.

## Upgrading from 6.7 to 7.0

There were two separate hiccups in the upgrade experience.

### Rolling upgrade affects sample dashboards

As I began the rolling upgrade, I started seeing "X of Y shards failed" errors in the System Overview dashboard. As the number of upgraded nodes increased, the reported number of failed shards increased.

![image](https://user-images.githubusercontent.com/1238659/54791459-359d0900-4bf7-11e9-8b1b-d2ea955ba504.png)

Once I had finished upgrading the last node, Kibana displayed a fatal error.

![image](https://user-images.githubusercontent.com/1238659/54791497-68df9800-4bf7-11e9-80d1-542c2c0d56e8.png)

Refreshing the browser allowed Kibana to load the dashboard successfully, but all panels were showing errors.

![image](https://user-images.githubusercontent.com/1238659/54791520-7d239500-4bf7-11e9-99f9-6f38aa755b49.png)

### Upgrading Metricbeat

To upgrade Logstash and Metricbeat, I killed both processes, configured Metricbeat with `migration.6_to_7.enabled: true`, set up the new 7.0.0 Metricbeat index template, and configured Logstash to rename documents so they would match the index template. Then I started ingesting data again.

Now, my dashboard had no errors but it wasn't loading data in many visualizations and it was showing shard failures again.

![image](https://user-images.githubusercontent.com/1238659/54791671-25d1f480-4bf8-11e9-941f-7c04673c8279.png)

ES logs showed this error:

```
[2019-03-21T16:49:58,044][DEBUG][o.e.a.s.TransportSearchAction] [1] [metricbeat-7.0.0-rc1-2019.03.21][3], node[Opsp4VRCTEWXJWEkDJArSQ], [P], s[STARTED], a[id=cHgnvBwoQ3yVv6TIhT5fdg]: Failed to execute [SearchRequest{searchType=QUERY_THEN_FETCH, indices=[metricbeat-*], indicesOptions=IndicesOptions[ignore_unavailable=true, allow_no_indices=true, expand_wildcards_open=true, expand_wildcards_closed=false, allow_aliases_to_multiple_indices=true, forbid_closed_indices=true, ignore_aliases=false, ignore_throttled=true], types=[], routing='null', preference='null', requestCache=null, scroll=null, maxConcurrentShardRequests=0, batchedReduceSize=512, preFilterShardSize=64, allowPartialSearchResults=true, localClusterAlias=null, getOrCreateAbsoluteStartMillis=-1, ccsMinimizeRoundtrips=true, source={"size":0,"timeout":"30000ms","query":{"bool":{"must":[{"query_string":{"query":"*","default_field":"*","fields":[],"type":"best_fields","default_operator":"or","max_determinized_states":10000,"enable_position_increments":true,"fuzziness":"AUTO","fuzzy_prefix_length":0,"fuzzy_max_expansions":50,"phrase_slop":0,"analyze_wildcard":true,"escape":false,"auto_generate_synonyms_phrase_query":true,"fuzzy_transpositions":true,"boost":1.0}},{"range":{"@timestamp":{"from":"2019-03-21T23:34:57.696Z","to":"2019-03-21T23:49:57.696Z","include_lower":true,"include_upper":true,"format":"strict_date_optional_time","boost":1.0}}}],"adjust_pure_negative":true,"boost":1.0}},"track_total_hits":2147483647,"aggregations":{"31e5afa1-1b1c-11e7-b09e-037021c4f8df":{"meta":{"timeField":"@timestamp","intervalString":"10s","bucketSize":10},"terms":{"field":"beat.name","size":10,"min_doc_count":1,"shard_min_doc_count":0,"show_term_doc_count_error":false,"order":[{"31e5afa2-1b1c-11e7-b09e-037021c4f8df-SORT":"desc"},{"_key":"asc"}]},"aggregations":{"31e5afa2-1b1c-11e7-b09e-037021c4f8df-SORT":{"avg":{"field":"system.memory.actual.used.pct"}},"timeseries":{"date_histogram":{"field":"@timestamp","time_zone":"America/Los_Angeles","interval":"10s","offset":0,"order":{"_key":"asc"},"keyed":false,"min_doc_count":0,"extended_bounds":{"min":1553211297696,"max":1553212197696}},"aggregations":{"31e5afa2-1b1c-11e7-b09e-037021c4f8df":{"avg":{"field":"system.memory.actual.used.pct"}}}}}}}}}]
org.elasticsearch.transport.RemoteTransportException: [4][127.0.0.1:9304][indices:data/read/search[phase/query]]
Caused by: org.elasticsearch.index.query.QueryShardException: failed to create query: {
  "bool" : {
    "must" : [
      {
        "query_string" : {
          "query" : "*",
          "default_field" : "*",
          "fields" : [ ],
          "type" : "best_fields",
          "default_operator" : "or",
          "max_determinized_states" : 10000,
          "enable_position_increments" : true,
          "fuzziness" : "AUTO",
          "fuzzy_prefix_length" : 0,
          "fuzzy_max_expansions" : 50,
          "phrase_slop" : 0,
          "analyze_wildcard" : true,
          "escape" : false,
          "auto_generate_synonyms_phrase_query" : true,
          "fuzzy_transpositions" : true,
          "boost" : 1.0
        }
      },
      {
        "range" : {
          "@timestamp" : {
            "from" : null,
            "to" : null,
            "include_lower" : true,
            "include_upper" : true,
            "boost" : 1.0
          }
        }
      }
    ],
    "adjust_pure_negative" : true,
    "boost" : 1.0
  }
}
	at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:309) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:292) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService.parseSource(SearchService.java:754) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService.createContext(SearchService.java:607) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:582) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:385) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService.access$100(SearchService.java:123) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:357) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:353) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService$4.doRun(SearchService.java:1068) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:41) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:751) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.IllegalArgumentException: field expansion matches too many fields, limit: 1024, got: 2252
	at org.elasticsearch.index.search.QueryParserHelper.checkForTooManyFields(QueryParserHelper.java:161) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.index.search.QueryParserHelper.resolveMappingField(QueryParserHelper.java:154) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.index.search.QueryParserHelper.resolveMappingField(QueryParserHelper.java:105) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.index.search.QueryStringQueryParser.<init>(QueryStringQueryParser.java:140) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.index.query.QueryStringQueryBuilder.doToQuery(QueryStringQueryBuilder.java:849) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:99) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.index.query.BoolQueryBuilder.addBooleanClauses(BoolQueryBuilder.java:394) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.index.query.BoolQueryBuilder.doToQuery(BoolQueryBuilder.java:378) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:99) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.index.query.QueryShardContext.lambda$toQuery$1(QueryShardContext.java:293) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:305) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:292) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService.parseSource(SearchService.java:754) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService.createContext(SearchService.java:607) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:582) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:385) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService.access$100(SearchService.java:123) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:357) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:353) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.search.SearchService$4.doRun(SearchService.java:1068) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:41) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:751) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-7.0.0-rc1.jar:7.0.0-rc1]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:834) ~[?:?]
```

**Comment by cjcenizal** [comment]
@ruflin @narph @fearful-symmetry @joshdover Unfortunately, the ideas we had from this morning did not resolve the problem. When upgrading from 6.5 to 6.7, I manually deleted the 6.5 dashboards and installed the sample dashboards for 6.7 by setting `setup.dashboards.enabled: true` in the Metricbeat config.

As I performed the ES rolling upgrade, the dashboard spat out shard failure notifications, the same as before. This screenshot is from 7.0.0 but it's still basically what I saw in 6.7.0:

![image](https://user-images.githubusercontent.com/1238659/54857070-bbce5380-4cba-11e9-91aa-bcdc43a98d77.png)

I also ran the query below on both 6.7 and 7.0 indices:

```
GET metricbeat-index/_search
{
    "query": {
        "query_string" : {
            "default_field": "*",
            "query" : "*" 
        }
    }
}
```

In both cases I got the "field expansion matches too many fields" error.

As a last resort I deleted the 6.7 dashboards and installed the 7.0 dashboards and of course those worked fine.

![image](https://user-images.githubusercontent.com/1238659/54857415-7448c700-4cbc-11e9-94dd-8b62d8482eb4.png)

**Comment by ruflin** [comment]
Why did you specify about `default_field: *`? I think this is expected to break.

I did some digging in our code base around 7.0 dashboards if we set the `default_field` still somewhere. It shows up in auditbeat (@andrewkroh @adriansr could have a look at this https://github.com/elastic/beats/blob/7.0/auditbeat/module/file_integrity/_meta/kibana/7/dashboard/auditbeat-file-integrity.json#L15 )? But that should not be relevant here I'm thinking.

@cjcenizal I assume you applied the magic from @joshdover to get the default_field setting into the old indices?

**Comment by cjcenizal** [comment]
@ruflin TBH I'm not sure why I specified `default_field` -- I don't know much about `default_field`, but I thought that was the query we all decided to use during the Zoom, so I kept it? Let me know if I should be testing with something else.

In terms of magic, I ran the 7.0 Upgrade Assistant which allows me to "fix" the 6.5 index's `default_fields` error.

**Comment by ruflin** [comment]
The query to be run should be with `default_field: *` as with this you overwrite the default_field setting. The default is what we want to test.

The Upgrade Assistant applies default_field setting to existing 6.5 indices. 

@cjcenizal In short, you still see errors with the most recent versions?

**Comment by cjcenizal** [comment]
## Metricbeat index pattern

I verified that the Kibana index pattern is `metricbeat-*`.

## default_fields and Upgrade Assistant

I've verified that the Upgrade Assistant adds `default_field` to the 6.5.4 index as we expect it to.

First, I verified the state of the indices, which corroborates the warnings surfaced by the Upgrade Assistant.

![image](https://user-images.githubusercontent.com/1238659/55117075-c025b280-50a6-11e9-9790-43cdcffe3a86.png)

![image](https://user-images.githubusercontent.com/1238659/55117079-c320a300-50a6-11e9-961d-3ede29511ff0.png)

![image](https://user-images.githubusercontent.com/1238659/55117080-c4ea6680-50a6-11e9-9cc7-6194fd14bd76.png)

After clicking the "Fix" button in Upgrade Assistant, the `default_field` property is added to the 6.5.4 index.

![image](https://user-images.githubusercontent.com/1238659/55117102-d6cc0980-50a6-11e9-812b-536b62ef13d7.png)

![image](https://user-images.githubusercontent.com/1238659/55117105-d92e6380-50a6-11e9-9742-b730a35b60e8.png)

## More information

I tried stopping both LS and Metricbeat before performing the above steps, I still get the same shard failure notifications and ES log errors as originally reported as soon as I begin the rolling upgrade.

If I run `metricbeat-*/_search`, I get results that increase in count as more logs are ingested, so the issue doesn't seem to lie with ES.

![image](https://user-images.githubusercontent.com/1238659/55119420-097a0000-50af-11e9-8386-4d01ffcf941e.png)

I ran the test queries on the 6.5.4 and 6.7.0 indices, and both execute without errors.

![image](https://user-images.githubusercontent.com/1238659/55119440-1565c200-50af-11e9-8c09-11d76c8f9544.png)

![image](https://user-images.githubusercontent.com/1238659/55119429-0e3eb400-50af-11e9-985d-0d24b33410af.png)

Refreshing the Kibana index pattern has no effect, except to show that `metricbeat-*` has mapping conflicts.

![image](https://user-images.githubusercontent.com/1238659/55117186-227eb300-50a7-11e9-93cb-fd8f0862af72.png)

At this point I wonder if there is an error with the index pattern and the way it's retrieving data.

What's _really_ weird is that it looks like the visualizations have just stopped working. The requests execute just fine, and I'm not seeing any ES errors in the logs but the visualizations are empty.

![image](https://user-images.githubusercontent.com/1238659/55119740-44c8fe80-50b0-11e9-8bb7-bc368e3f0726.png)

Here's the response, containing data:

```json
{
  "took":3,
  "responses":[
    {
      "took":3,
      "timed_out":false,
      "_shards":{
        "total":16,
        "successful":16,
        "skipped":0,
        "failed":0
      },
      "hits":{
        "total":1353,
        "max_score":null,
        "hits":[

        ]
      },
      "aggregations":{
        "2":{
          "buckets":[
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:36:00.000-07:00",
              "key":1553729760000,
              "doc_count":33
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:36:30.000-07:00",
              "key":1553729790000,
              "doc_count":81
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:37:00.000-07:00",
              "key":1553729820000,
              "doc_count":88
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:37:30.000-07:00",
              "key":1553729850000,
              "doc_count":84
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:38:00.000-07:00",
              "key":1553729880000,
              "doc_count":91
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:38:30.000-07:00",
              "key":1553729910000,
              "doc_count":83
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:39:00.000-07:00",
              "key":1553729940000,
              "doc_count":89
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:39:30.000-07:00",
              "key":1553729970000,
              "doc_count":83
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:40:00.000-07:00",
              "key":1553730000000,
              "doc_count":93
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:40:30.000-07:00",
              "key":1553730030000,
              "doc_count":85
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:41:00.000-07:00",
              "key":1553730060000,
              "doc_count":92
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:41:30.000-07:00",
              "key":1553730090000,
              "doc_count":84
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:42:00.000-07:00",
              "key":1553730120000,
              "doc_count":88
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:42:30.000-07:00",
              "key":1553730150000,
              "doc_count":84
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:43:00.000-07:00",
              "key":1553730180000,
              "doc_count":89
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:43:30.000-07:00",
              "key":1553730210000,
              "doc_count":83
            },
            {
              "3":{
                "doc_count_error_upper_bound":0,
                "sum_other_doc_count":0,
                "buckets":[

                ]
              },
              "key_as_string":"2019-03-27T16:44:00.000-07:00",
              "key":1553730240000,
              "doc_count":23
            }
          ]
        }
      },
      "status":200
    }
  ]
}
```

**Comment by ruflin** [comment]
Thanks for all the detailed testing and sharing the path. First a note on the index pattern: Do not refresh it as this will overwrite the settings provided by Beats. The Kibana index pattern should never be touched manually. The warning you see I think is correct as indeed between 6.7 and 7.0 there are changes in some existing fields, but it should not affect any of the queries. Did you refersh it manually also earlier?

The part in the above I don't really understand is: Dashboard does not show data but when running the query, there is data? My first thinking was if it is perhaps something with the date range / data that it has an offset?

For the upgrade error we see, I'm wondering who runs this query. Did you have Kibana open during the migration? Otherwise I'm thinking it could be the upgrade assitant which then adds the fields that does run such a check. @joshdover might know more here. 
Does Elasticsearch continue normally afterwards?

**Comment by cjcenizal** [comment]
> Did you refresh it manually also earlier?

No, I refreshed as a last resort.

>  Dashboard does not show data but when running the query, there is data? My first thinking was if it is perhaps something with the date range / data that it has an offset?

Yes, it's very weird. What's happening is the visualization doesn't show any data, but the visualization inspector shows that ES is responding with data. I've pinged the Kibana App team to try to get some help digging into this.

> For the upgrade error we see, I'm wondering who runs this query. Did you have Kibana open during the migration?

I believe the query is caused by the Metricbeat visualization refreshing (yes, Kibana is open).

**Comment by cjcenizal** [comment]
So I've managed to define two different issues we're having.

## 6.7 Kibana Advanced Settings sets `{ "default_field": "*" }` 

In 6.7 Kibana, there is an Advanced Settings option called "Query string options" which defaults to `{  "default_field": "*" }`, which is then applied to every request. This was causing the shard failures and ES log errors I originally reported. Removing this setting solves the problem. I'm going to take further steps on this:

1. I'll verify that this problem _doesn't_ occur when you install 6.7 fresh (as opposed to upgrade from 6.5) and try to figure out why.
2. I'll take with the Kibana team to establish the best solution for this. Some options: remove this default in 6.7.1, fix this problem in Upgrade Assistant in 6.7.1, or add steps for fixing this problem in our upgrade docs.

## The Logstash docs provide incorrect `output` guidance

The [Getting started with Logstash docs](https://www.elastic.co/guide/en/logstash/current/advanced-pipeline.html#indexing-parsed-data-into-elasticsearch) advise the user to set up their `output` incorrectly. It should provide the same guidance as in the [Beats docs](https://www.elastic.co/guide/en/beats/filebeat/current/logstash-output.html), which advises to dynamically set the index name.

Users who have followed the docs as I have will inadvertently end up with a Logstash pipeline that sets the wrong index name, which might not be caught by the 7.0 Metricbeat index template. In this case, their index will be missing default fields and aliases, and it will look like their Metricbeat visualizations and dashboards are no longer loading new data.

@ruflin will own next steps:

1. Fixing the Logstash docs by bringing them inline with the Beats docs.
2. Providing trouble-shooting steps for identifying and fixing this problem in the 7.0 stack upgrade docs.

**Comment by cjcenizal** [comment]
@ruflin Might it make sense to also change the Metricbeat index template to have a greedier index pattern that's not coupled with the version? E.g., `metricbeat-*`?

**Comment by cjcenizal** [comment]
## 6.7 Kibana Advanced Settings investigation

When I start a fresh 6.7 installation, I see the `default_field: *` query in the visualization inspector. There are no shard failures or errors in the ES log. I'm also able to run this query in Console without any problem.

![image](https://user-images.githubusercontent.com/1238659/55197248-3988d800-516f-11e9-8d8e-452ca3aac690.png)

![image](https://user-images.githubusercontent.com/1238659/55197238-31c93380-516f-11e9-8060-acaf8f3c5236.png)

However, once I upgrade the cluster to 7.0.0, I start seeing the same shard failures and ES errors I saw before. It seems this problem only occurs during the upgrade process, while Kibana is on 6.7 and ES is on 7.0.

![image](https://user-images.githubusercontent.com/1238659/55197314-6937e000-516f-11e9-9b3e-ffd1f79d04c1.png)

One solution is to remove `default_field` from Advanced Settings. As expected, this fixes the problem.

![image](https://user-images.githubusercontent.com/1238659/55197341-7c4ab000-516f-11e9-849c-6207cda4e307.png)

Another solution is to continue the stack upgrade process. Once Kibana is upgraded to 7.0, it updates the Advanced Setting and removes `default_field` for you.

![image](https://user-images.githubusercontent.com/1238659/55197361-8e2c5300-516f-11e9-890c-43a86e773b22.png)

**Comment by cjcenizal** [comment]
I was able to confirm that if I just _ignore_ all of the shard failures and ES errors that pop up during the upgrade process due to `default_fields`, and if I deliberately avoid the Logstash output configuration problem noted above, then once I am done upgrading everything to 7.0 everything looks 💯. The 6.5 dashboards continue to update with new data as it's being ingested, without any shard failures or ES errors in the log.

**Comment by andrewkroh** [comment]
I opened an issue to track removing `default_field` from the Auditbeat dashboards. https://github.com/elastic/beats/issues/11568

**Comment by joshdover** [comment]
> I'll talk with the Kibana team to establish the best solution for this. Some options: remove this default in 6.7.1, fix this problem in Upgrade Assistant in 6.7.1, or add steps for fixing this problem in our upgrade docs.

@cjcenizal Is there a Kibana issue open for discussing changes needed for 6.7.x?

**Comment by cjcenizal** [comment]
@joshdover Just created it for ya: https://github.com/elastic/kibana/issues/34360
